Added interface "DataStorage", added class "FileDataStorage"

RakaSamaAi opened the pull request.

RakaSamaAi committed:
add Test.java code ^_^

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java

Liberate520 commented on src/Person.java:
```diff
@@ -0,0 +1,35 @@
+
+public class Person {
+    private String name;
+    private int age;
```
Правильно будет дата рождения

Liberate520 commented on src/Person.java:
```diff
@@ -0,0 +1,35 @@
+public class Person {
+    private String name;
+    private int age;
+    private List<Person> children;
```
Рекомендую также добавить ссылки на родителей. В таком проекте это очень актуальное поле

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages

RakaSamaAi committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/FamilyTree.java
#	src/Main.java
#	src/model/Person.java

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages
Added by gitignor

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages
Added by gitignor
add CommandProcessor.java and changed main

Liberate520 commented on src/model/FamilyTree.java:
```diff
@@ -0,0 +1,41 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class FamilyTree<T> {
```
По заданию класс должен реализовать интерфейс Iterable

Liberate520 replied:
параметр надо ограничить, как мы делали это на семинаре. Создать для этого отдельный интерфейс

Liberate520 commented on src/model/FamilyTree.java:
```diff
@@ -0,0 +1,41 @@
+        }
+        return null;
+    }
+
```
А также содержать методы сортировки

Liberate520 commented on src/model/FamilyTree.java:
```diff
@@ -0,0 +1,41 @@
+        this.members.add(person);
+    }
+
+    public List<Person> getChildrenOf(Person parent) {
```
Не должно быть упоминания класс Person, если уж вы хотите написать класс в общем виде. Заменить на параметр

Liberate520 commented on src/model/FamilyTree.java:
```diff
@@ -0,0 +1,41 @@
+
+    public T findPersonByName(String name) {
+        for (T person : members) {
+            if (person instanceof Person && ((Person) person).getName().equals(name)) {
```
Приведения типов в данном случае это ошибка. Идея обобщения в том, чтобы работало с любым типом данных. Например, если мы будем использовать для составления родословной собак

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages
Added by gitignor
add CommandProcessor.java and changed main
Added new pockets and interfaces for the pattern MVP

Liberate520 commented on src/presenter/FamilyTreePresenter.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.List;
+
+public class FamilyTreePresenter implements TreePresenter {
+    private FamilyTree<Person> familyTree;
+    private TreeView view;
+    private DataStorage storage;
```
Надо добавить класс Сервис для объединения работы модели. Сервис уже будет управлять классами FamiltyTree и DataStorage

Liberate520 commented on src/view/ConsoleTreeView.java:
```diff
@@ -0,0 +1,86 @@
+            System.out.println("Enter command (add/find/children/exit):");
+            String command = scanner.nextLine();
+
+            switch (command) {
+                case "add":
+                    addMember();
+                    break;
+                case "find":
+                    findMember();
+                    break;
+                case "children":
+                    getChildrenOf();
+                    break;
+                case "exit":
+                    return;
```
будет нарушать второй принцип солид

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages
Added by gitignor
add CommandProcessor.java and changed main
Added new pockets and interfaces for the pattern MVP
I redid almost everything

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages
Added by gitignor
add CommandProcessor.java and changed main
Added new pockets and interfaces for the pattern MVP
I redid almost everything
small alteration

RakaSamaAi committed:
I deleted the test and put "Main" because I mixed it up :)
add FamilyTree.java
add Person.java
Added interface "DataStorage", added class "FileDataStorage"
Added Iterable and divided it into packages
Added by gitignor
add CommandProcessor.java and changed main
Added new pockets and interfaces for the pattern MVP
I redid almost everything
small alteration
change main and CommandProcessor

Liberate520 commented on src/presenter/FamilyTreePresenter.java:
```diff
@@ -0,0 +1,69 @@
+
+    @Override
+    public void addMember(String name, int age) {
+        Person person = new Person(name, age);
```
презентер не должен создавать объекты модели

Liberate520 commented on src/presenter/FamilyTreePresenter.java:
```diff
@@ -0,0 +1,69 @@
+            view.showMessage("Человек не найден.");
+        }
+    }
+
```
потеряна возможность создания родственных связей
